event_callback.go

```diff
@@ -0,0 +1,170 @@
+package firego
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/zabawaba99/firego/sync"
+)
+
+// ChildEventFunc is the type of function that is called for every
+// new child added under a firebase reference. The snapshot argument
+// contains the data that was added. The previousChildKey argument
+// contains the key of the previous child that this function was called for.
+type ChildEventFunc func(snapshot DataSnapshot, previousChildKey string)
+
+// ChildAdded listens on the firebase instance and executes the callback
+// for every child that is added.
+func (fb *Firebase) ChildAdded(fn ChildEventFunc) error {
+	handleSSE := func(notifications chan Event) {
```

Comment:
consider extracting this anonymous func as an unexported method, or as an attribute of `fb`.